File: java-sdk/lib/src/main/java/dev/formance/javasdk/models/shared/Security.java
package dev.formance.javasdk.models.shared;

import dev.formance.javasdk.utils.SpeakeasyMetadata;

public class Security {
    @SpeakeasyMetadata("security:scheme=true,type=oauth2")
    public SchemeAuthorization authorization;
    public Security withAuthorization(SchemeAuthorization authorization) {
        this.authorization = authorization;
        return this;
    }
}

File: ts-sdk/src/sdk/models/shared/security.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class SchemeAuthorization extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, name=Authorization" })
  authorization: string;
}

export class Security extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  authorization: SchemeAuthorization;
}